src/query/interpreter.cpp

```diff
@@ -1399,6 +1399,139 @@ PreparedQuery PrepareDumpQuery(ParsedQuery parsed_query, std::map<std::string, T
                        RWType::R};
 }
 
+std::vector<std::vector<TypedValue>> AnalyzeGraphQueryHandler::AnalyzeGraphCreateStatistics(
+    const std::span<std::string> labels, DbAccessor *execution_db_accessor) {
+  using LPIndex = std::pair<storage::LabelId, storage::PropertyId>;
+
+  std::vector<std::vector<TypedValue>> results;
+  std::map<LPIndex, std::map<storage::PropertyValue, int64_t>> counter;
+
+  // Preprocess labels to avoid later checks
+  std::vector<LPIndex> indices_info = execution_db_accessor->ListAllIndices().label_property;
+  if (labels[0] != kAsterisk) {
+    for (auto it = indices_info.cbegin(); it != indices_info.cend();) {
+      if (std::find(labels.begin(), labels.end(), execution_db_accessor->LabelToName(it->first)) == labels.end()) {
+        indices_info.erase(it);
+      } else {
+        ++it;
+      }
```

Comment:
How does this work? Should you not assign the new iterator after erasing, since the iterator you are holding is invalidated and you cannot increment on it anymore, e.g.
```suggestion
      if (std::find(labels.begin(), labels.end(), execution_db_accessor->LabelToName(it->first)) == labels.end()) {
        it = indices_info.erase(it);
      } else {
        ++it;
      }
```

Reply:
I deleted this it accidentally, it should be like you wrote...